lldb/source/Commands/CommandObjectSettings.cpp

```diff
@@ -237,28 +237,62 @@ insert-before or insert-after.");
 };
 
 // CommandObjectSettingsShow -- Show current values
+#define LLDB_OPTIONS_settings_show
+#include "CommandOptions.inc"
 
 class CommandObjectSettingsShow : public CommandObjectParsed {
 public:
   CommandObjectSettingsShow(CommandInterpreter &interpreter)
       : CommandObjectParsed(interpreter, "settings show",
                             "Show matching debugger settings and their current "
-                            "values.  Defaults to showing all settings.",
-                            nullptr) {
+                            "values.  Defaults to showing all settings.") {
     AddSimpleArgumentList(eArgTypeSettingVariableName, eArgRepeatOptional);
   }
 
   ~CommandObjectSettingsShow() override = default;
 
+  Options *GetOptions() override { return &m_options; }
+
+  class CommandOptions : public Options {
+  public:
+    ~CommandOptions() override = default;
+
+    Status SetOptionValue(uint32_t option_idx, llvm::StringRef option_arg,
+                          ExecutionContext *execution_context) override {
+      const int short_option = m_getopt_table[option_idx].val;
+      switch (short_option) {
+      case 'v':
```

Comment:
This is a tiny quibble, but unless we have other things in mind to include in a verbose listing, I'd suggest `--show-default` and `-d` rather than `--verbose` since that way we say exactly what we're doing.  I can't think of anything else we'd include here, so `--show-defaults` seems better to me.

Reply:
And even if we do add extra stuff, seeing what the defaults are on their own seems like a useful enough action I expect people would want to dial it up on its own.

Reply:
what do you think of `--include-defaults` since "show" might imply that a default will be printed for every setting, but that's not what the PR currently does. Or are you also suggesting that this should be changed to show a default, even for cases where for example a setting type doesn't have a default value (array, dict, etc).

Reply:
If we were to do separate displays for "defaults", "values" and "values  with defaults" then show would be appropriate.  But I think it's fine not to do that.  As you said below, you can run w/o an .lldbinit in command line to figure that out, and in Xcode you could do the same thing by making an empty target-specific lldb init file.  Presumably there's a way to do this in VSCode, but even if there weren't this is a special-purpose enough task having to do it with `-x` seems acceptable.

In that case, `--include-defaults` seems more accurate.